### PR TITLE
feat(profiling): add DD_PROFILING_HEAP_ENABLED env var to enable heap profiler

### PR DESCRIPTION
**Pull request by jd**

### CODE REVIEW

**Review by P403n1x87 — COMMENTED** [review]
Can we add a test case similar to #2567 for `ddtrace-run`?

File: docs/configuration.rst [hunk 1]
```diff
@@ -109,6 +109,10 @@ below:
      - Integer
      - 64
      - The maximum number of frames to capture in stack execution tracing.
+   * - ``DD_PROFILING_HEAP_ENABLED``
+     - Boolean
+     - 0
```
Comment (Kyle-Verhoog) [inline docs/configuration.rst:114]:
```suggestion
     - False
```

Seems to be our convention 🤷‍♂️

**Review by Kyle-Verhoog — APPROVED** [review]
yay!

**Comment by jd** [comment]
> Can we add a test case similar to #2567 for ddtrace-run?

We can, I'm just not sure how worth it is, considering the env var is temporary and would get removed ~next-nextrelease.
WDYT?

**Review by Kyle-Verhoog — APPROVED** [review]

**Comment by jd** [comment]
@P403n1x87 I've looked into adding end-to-end test but honestly it's going to be too complicated to test for the value it adds at this point.

**Review by nizox — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**